### ISSUE TITLE
Error: "Cannot read properties of undefined (reading 'id')" when opening the app for the first time

### ISSUE THREAD

**Issue by Malte-BNT** [OfficeDev/microsoft-teams-apps-champion-management#194]
As SharePoint Admin, we installed CMP v2.5 to the tenant-wide SharePoint App catalog, enabled it for all Sites and Teams (and published to Teams), then approved the required API permissions as Global Admin.
Teams App "Champion Management Platform" is available for install, so a non-admin user installed it in personal scope. When opening the app (for the first time), it only renders an error message (see below). That said, we are not getting to the initial screen with the spinner (while site/lists get created).

```
Something went wrong
If the problem persists, contact the site administrator and give them the information in Technical Details.
TECHNICAL DETAILS
ERROR:
Cannot read properties of undefined (reading 'id')

CALL STACK:
TypeError: Cannot read properties of undefined (reading 'id')
    at https://res-1.cdn.office.net/files/sp-client/teams-web-part-application-assembly_en-us_b2479fff0532f5aec35976ff8e0b39fe.js:125:12346
    at new Promise (<anonymous>)
    at e.loadComponent (https://res-1.cdn.office.net/files/sp-client/teams-web-part-application-assembly_en-us_b2479fff0532f5aec35976ff8e0b39fe.js:125:12314)
    at https://res-1.cdn.office.net/files/sp-client/teams-web-part-application-assembly_en-us_b2479fff0532f5aec35976ff8e0b39fe.js:125:14535
    at async Promise.all (index 0)
```

The same happens when opening the app in either Teams Web Client (browser: https://teams.microsoft.com) and SharePoint (adding the app as a web part to a page).
At least on SharePoint, we tried again with a user account being a SharePoint Admin. We can't use Teams with that same account - no Teams license assigned. Not sure if this is a requirement, but to my understanding, it is not (users are allowed to create SharePoint sites through API, script, etc.).

The web version of the Teams Client exposed several stack traces, I attached those (replacing and tenant references with "*removed*).
[teams.microsoft.com-1712243717399.log](https://github.com/OfficeDev/microsoft-teams-apps-champion-management/files/14873454/teams.microsoft.com-1712243717399.log)

Hope that helps you to guide us where to look at.

Thanks, Malte

**Comment by v-saikirang** [comment]
Hi @Malte-BNT,

Thanks for reaching out to us.

As mentioned in the deployment guide, please try opening the app with a global admin account. If you still face issues please check the below:

1) check if SharePoint site "championmanagementplatform" is created.
2) please share screenshot the error you see when running the app.
3) share a screenshot of the API permissions page.
4) if the SharePoint site is created share a screenshot of the site contents page

Thanks.

**Comment by Malte-BNT** [comment]
Hi @v-saikirang,

thanks for the fast reply.

I installed the Teams App with a user being Global Admin now, but still: when opening the App with that Global Admin user account, it displays the above error message (screenshot attached as well, now) before rendering anything.

I checked the following (screenshots attached):

1. SharePoint site "championmanagementplatform" is not created (searched for it in SharePoint Admin Center)
2. Screenshot of the error message attached
3. SharePoint API permission page (screenshot attached)
4. no site created yet

![image](https://github.com/OfficeDev/microsoft-teams-apps-champion-management/assets/102792039/fb15104f-beee-49fc-aff3-bc690a5fe15d)
![image](https://github.com/OfficeDev/microsoft-teams-apps-champion-management/assets/102792039/43c77163-1b6f-482b-b277-c1b39528bdbd)
![image](https://github.com/OfficeDev/microsoft-teams-apps-champion-management/assets/102792039/5bb6ad31-0d9b-456c-9c5b-1e5d0fbcf54a)
![image](https://github.com/OfficeDev/microsoft-teams-apps-champion-management/assets/102792039/3ebfadda-cd58-499c-a58b-a8f3f63093dd)

Malte
